app/jobs/client_cleanup_job.rb

```diff
@@ -0,0 +1,46 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+class ClientCleanupJob < BaseJob
+  queue_as ENV.fetch('DJ_LONG_QUEUE_NAME', :long_running)
+  queue_with_priority 6
+
+  WAIT_MINUTES = 1
+
+  def perform(ids)
+    running_jobs = Delayed::Job.jobs_for_class('ClientCleanupJob').running.count
+    if running_jobs > 1
+
+      requeue_job
+      return
+    end
+
+    GrdaWarehouse::Tasks::ClientCleanup.run_for_clients(ids)
+  end
```

Comment:
consider if we can just synchronize with and advisory lock to avoid race conds
```suggestion
  def perform(ids)
    lock_obtained = nil
    lock_name = self.class.name.to_s
    with_advisory_lock(lock_name, timeout_seconds: 0) do
      lock_obtained = true
      GrdaWarehouse::Tasks::ClientCleanup.run_for_clients(ids)
    end
    requeue_job unless lock_obtained
  end
```